arviz/tests/external_tests/test_data_pyjags.py

```diff
@@ -0,0 +1,99 @@
+# pylint: disable=redefined-outer-name
+
+import typing as tp
+import numpy as np
+import pytest
+
+from arviz.data.io_pyjags import (
+    _convert_pyjags_dict_to_arviz_dict,
+    _convert_arviz_dict_to_pyjags_dict,
+    _extract_arviz_dict_from_inference_data,
+    from_pyjags,
+)
+
+from arviz.tests.helpers import check_multiple_attrs
+
+
+@pytest.fixture()
+def pyjags_samples_dict() -> tp.Mapping[str, np.ndarray]:
+    return {"b": np.random.randn(3, 10, 3), "int": np.random.randn(1, 10, 3)}
+
+
+# pyjags_samples_dict = {
+#         "b": np.random.randn(3, 10, 3),
+#         "int": np.random.randn(1, 10, 3)
+#     }
+
+
+def verify_equality_of_numpy_values_dictionaries(
+    dict_1: tp.Mapping[tp.Any, np.ndarray], dict_2: tp.Mapping[tp.Any, np.ndarray]
+) -> bool:
+    if dict_1.keys() != dict_2.keys():
+        return False
+
+    for key in dict_1.keys():
+        if not np.all(dict_1[key] == dict_2[key]):
+            return False
+
+    return True
+
+
+def test_convert_pyjags_samples_dictionary_to_arviz_samples_dictionary(
+    pyjags_samples_dict: tp.Mapping[str, np.ndarray],
+):
+    arviz_samples_dict_from_pyjags_samples_dict = _convert_pyjags_dict_to_arviz_dict(
+        pyjags_samples_dict
+    )
+
+    pyjags_dict_from_arviz_dict_from_pyjags_dict = _convert_arviz_dict_to_pyjags_dict(
+        arviz_samples_dict_from_pyjags_samples_dict
+    )
+
+    assert verify_equality_of_numpy_values_dictionaries(
+        pyjags_samples_dict, pyjags_dict_from_arviz_dict_from_pyjags_dict,
+    )
+
+
+def test_extract_samples_dictionary_from_arviz_inference_data(pyjags_samples_dict):
+    arviz_samples_dict_from_pyjags_samples_dict = _convert_pyjags_dict_to_arviz_dict(
+        pyjags_samples_dict
+    )
+
+    arviz_inference_data_from_pyjags_samples_dict = from_pyjags(pyjags_samples_dict)
+    arviz_dict_from_idata_from_pyjags_dict = _extract_arviz_dict_from_inference_data(
+        arviz_inference_data_from_pyjags_samples_dict
+    )
+
+    assert verify_equality_of_numpy_values_dictionaries(
+        arviz_samples_dict_from_pyjags_samples_dict, arviz_dict_from_idata_from_pyjags_dict,
+    )
+
+
+def test_roundtrip_from_pyjags_via_arviz_to_pyjags(
+    pyjags_samples_dict: tp.Mapping[str, np.ndarray],
+):
+    arviz_inference_data_from_pyjags_samples_dict = from_pyjags(pyjags_samples_dict)
+    arviz_dict_from_idata_from_pyjags_dict = _extract_arviz_dict_from_inference_data(
+        arviz_inference_data_from_pyjags_samples_dict
+    )
+
+    pyjags_dict_from_arviz_idata = _convert_arviz_dict_to_pyjags_dict(
+        arviz_dict_from_idata_from_pyjags_dict
+    )
+
+    assert verify_equality_of_numpy_values_dictionaries(
+        pyjags_samples_dict, pyjags_dict_from_arviz_idata
+    )
+
+
+@pytest.mark.parametrize("warmup_iterations", [0, 5])
+def test_inference_data_attrs(pyjags_samples_dict, warmup_iterations: int):
```

Comment:
divide this test into a couple or 3 tests.  

1. `test_from_pyjags` same code as here removing `warmup_iterations` and adding also prior group
2. `test_from_pyjags_warmup` same code as here (prior not necessary), test warmup inclusion. I am not sure what cases are interesting though, warmup_iterations > 0 and save_warmup = True for sure, but warmup_iterations > 0 and save_warmup = False? maybe that should be an error or a warning? and iter=0 + True?
3. `test_from_pyjags_warning/error`? (to be added depending on what we decide regarding the question above)

This here I think is the main unresolved issue.

Reply:
Not sure I correctly understand the intention behind everything. 

1.) I added prior samples. I am now including all the cases with and without prior/posterior. I am not sure creating a test-case without the argument warmup_iterations is required if we include a case where we set the parameter to zero, since it defaults to zero anyway. Once again, I am not sure if I understood what you meant.

2.) I created all of the combinations:
    - save_warmup True/False/None
    - warmup_iterations: 0, 5
    The combination warmup_iterations=5 and save_warmup=False is legitimate in my opinion. Perhaps the warmup iterations use a lot of memory and the user decides not to keep them. But it is still important for the converter to not include the warmup iterations for subsequent analysis using ArviZ.

   The combination of save_warmup = True and warmup_iterations = False simply returns an empty dictionary. We could also return a dictionary of empty NumPy arrays. 

What do you think?

Reply:
Thanks for all the explanation, everything makes sense. Returning an empy dictionary is perfect and would make `InferenceData` initializer to not create the group at all (which is what we want, the other converters do not create empty groups on purpose)

Reply:
Thanks for following up. Glad we are on the same page.